### PR TITLE
Move TCL test unit/type/set to Go case

### LINKED ISSUE #819: Move TCL test unit/type/set to Go case

### PR DESCRIPTION
**Pull request by tanruixiang**
it closes #819

### CODE REVIEW

**Comment by tanruixiang** [comment]
Hi. I found that the CI failure was accidental, and it seemed that it had nothing to do with my modification. [fail 1](https://github.com/apache/incubator-kvrocks/runs/8297760486?check_suite_focus=true) [fail 2](https://github.com/apache/incubator-kvrocks/runs/8294594464?check_suite_focus=true) This two failure's code is the same(just delete some empty lines),but fail in different CI and different test.

**Comment by tanruixiang** [comment]
@tisonkun @git-hulk  @PragmaTwice  HI.  I found that the results of CI were a bit random. My same code failed in different places the first two times, and I passed it the last time.

File: tests/gocase/unit/type/set/set_test.go [hunk 1]
```diff
@@ -0,0 +1,741 @@
...
+		})
+
+		t.Run("SUNION with two sets - "+dstype, func(t *testing.T) {
+			set1 := rdb.SMembers(ctx, "set1")
+			require.NoError(t, set1.Err())
+			set2 := rdb.SMembers(ctx, "set2")
+			require.NoError(t, set2.Err())
+			var expect []string
+			var vis = make(map[string]bool)
+			for _, i := range set1.Val() {
```
Comment (git-hulk) [inline tests/gocase/unit/type/set/set_test.go]: I saw this unique merge code block appears many times, do you mind adding a util function for merging N sets?

**Review by git-hulk — COMMENTED** [review]
Thanks for your great contribution again.

**Comment by git-hulk** [comment]
> a bit random. My same code failed in different places the first two times, and I passed it the last time.

Yes, there's some tcl random failures but we can't produce them locally, so can just ignore them.

**Comment by tisonkun** [comment]
> some tcl random failures

I notice they're all for `scripting.tcl`. If you found any failure for others, you can point it out. I hope when migrating tests to Go cases, we can dig a bit to prevent new unstable tests.

File: tests/gocase/unit/type/set/set_test.go (hunk 1 above)
Comment (tanruixiang) [reply tests/gocase/unit/type/set/set_test.go]: ok. I have added it.

Comment (git-hulk) [reply tests/gocase/unit/type/set/set_test.go]: Thanks

**Review by git-hulk — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]
@tanruixiang BTW, I'll be away for a few days. You may not expect my reviews in time for other patches. You can always ask for other committers' reviews :)

**Comment by tisonkun** [comment]
```
BenchmarkLTRIM
BenchmarkLTRIM/LTRIM_stress_testing_-_zipList
BenchmarkLTRIM/LTRIM_stress_testing_-_zipList-3         	       1	10113128988 ns/op
BenchmarkLTRIM/LTRIM_stress_testing_-_linkedList
    list_test.go:80: 
        	Error Trace:	/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/type/list/list_test.go:80
        	            				/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/type/list/benchmark.go:193
        	            				/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/type/list/benchmark.go:233
        	            				/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/type/list/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	i/o timeout
        	Test:       	BenchmarkLTRIM/LTRIM_stress_testing_-_linkedList
--- FAIL: BenchmarkLTRIM/LTRIM_stress_testing_-_linkedList
--- FAIL: BenchmarkLTRIM
FAIL
```

Seems a new unstable test.

**Comment by git-hulk** [comment]
> ```
> BenchmarkLTRIM
> BenchmarkLTRIM/LTRIM_stress_testing_-_zipList
> BenchmarkLTRIM/LTRIM_stress_testing_-_zipList-3         	       1	10113128988 ns/op
> BenchmarkLTRIM/LTRIM_stress_testing_-_linkedList
>     list_test.go:80: 
>         	Error Trace:	/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/type/list/list_test.go:80
>         	            				/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/type/list/benchmark.go:193
>         	            				/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/type/list/benchmark.go:233
>         	            				/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/type/list/asm_amd64.s:1594
>         	Error:      	Received unexpected error:
>         	            	i/o timeout
>         	Test:       	BenchmarkLTRIM/LTRIM_stress_testing_-_linkedList
> --- FAIL: BenchmarkLTRIM/LTRIM_stress_testing_-_linkedList
> --- FAIL: BenchmarkLTRIM
> FAIL
> ```
> 
> Seems a new unstable test.

Looks like bench test may cause Kvrocks high latency and then client got timeout error.

**Comment by tisonkun** [comment]
@git-hulk see #870 .

**Comment by tanruixiang** [comment]
> @tanruixiang BTW, I'll be away for a few days. You may not expect my reviews in time for other patches. You can always ask for other committers' reviews :)

ok. thank you very much.

**Pull request merged**